File: Behavioral/Strategy/CSharp/BitcoinPayment.cs
// BitcoinPayment.cs

using System;

public class BitcoinPayment : IPaymentStrategy {
  private string _walletAddress;

  public BitcoinPayment(string walletAddress) {
    _walletAddress = walletAddress;
  }

  public void Pay(double amount) {
    Console.WriteLine($"Paid ${amount} using Bitcoin (Wallet: {_walletAddress})");
  }
}

File: Behavioral/Strategy/Cpp/BitcoinPayment.cpp
// BitcoinPayment.cpp

#include "BitcoinPayment.h"

BitcoinPayment::BitcoinPayment(const std::string& walletAddress) : walletAddress(walletAddress) {}

void BitcoinPayment::pay(double amount) {
  std::cout << "Paid $" << amount << " using Bitcoin (Wallet: " << walletAddress << ")" << std::endl;
}
